snap/validate.go

```diff
@@ -951,3 +951,43 @@ func ValidateSystemUsernames(info *Info) error {
 	}
 	return nil
 }
+
+// neededDefaultProviders returns the names of all default-providers for
+// the content plugs that the given snap.Info needs.
+func NeededDefaultProviders(info *Info) (cps []string) {
+	for _, plug := range info.Plugs {
+		if plug.Interface == "content" {
+			var dprovider string
+			if err := plug.Attr("default-provider", &dprovider); err == nil && dprovider != "" {
+				cps = append(cps, dprovider)
+			}
+		}
+	}
+	return cps
+}
+
+// ValidateBasesAndProviders checks that all bases/default-providers are part of the seed
+func ValidateBasesAndProviders(snapInfos map[string]*Info) []error {
```

Comment:
now that it's here, this probably needs some unit tests

Reply:
Added (+ we have the existing tests in image/validate_seed_test.go).